src/Microsoft.Data.Sqlite/Interop/VersionedMethods.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Data.Sqlite.Interop
+{
+    internal class VersionedMethods
```

Comment:
Could take this one step further with the strategy pattern. You'd set the strategy once in the static constructor based on the version then each method wouldn't have to check each time.